12.2 home

ATSchoolENikonov opened the pull request.

methshest committed:
12.3
промежуточный коммит

methshest committed:
12.3
промежуточный коммит

methshest committed:
12.2
исправления

methshest committed:
12.2
че-то получилось, осталось докрутить
Launch  чисто для отладки нужен был,я оставил

Separador commented on Home_12.2/src/main/java/MovieRepositoryImpl.java:
```diff
@@ -0,0 +1,98 @@
+    public MovieRepositoryImpl() {
+        try {
+            connection = DriverManager.getConnection("jdbc:postgresql://localhost:5432/postgres",
+                    "postgres",
```
В реальной жизни такого быть не должно. Логины и пароли должны быть либо зашифрованы, либо передаваться через окружение jvm.

Separador commented on Home_12.2/src/test/java/Task_2.java:
Это  для чего?

Separador commented on Home_12.2/src/test/java/Task_1.java:
Это  для чего?

Separador commented on Home_12.2/src/main/java/DirectorRepositoryImpl.java:
```diff
@@ -0,0 +1,107 @@
+
+    public DirectorRepositoryImpl() {
+        try {
+            connection = DriverManager.getConnection("jdbc:postgresql://localhost:5432/postgres",
+                    "postgres",
+                    "Polgen366");
```
В реальной жизни такого быть не должно. Логины и пароли должны быть либо зашифрованы, либо передаваться через окружение jvm.

Separador reviewed (changes requested):
1) По условию таблицы и необходимые данные должны добавляться, если отсутствуют, автоматически перед запуском тестов.
2) Используй try-with-resources для statement'ов
3) хорошо бы вынести создание соединения в отдельное место, чтобы эти настройки не дублировать каждый раз
4) вынеси строки запросов в константы и вместо конкатенации используй '?' для PreparedStatement'ов

Separador commented on Home_12.2/src/main/java/DirectorRepositoryImpl.java:
```diff
@@ -0,0 +1,107 @@
+                    "Select * from directors d join movies m on d.id=m.director where genre in (" + str + ");");
+            for (int i = 0; i < genres.size(); i++) {
+                statement.setString(count,genres.get(i));
+                count++;
```
Переменная кажется нигде не используется

methshest committed:
12.2
1)Добавил класс с коннектом и созданием тестданных
2)Сделал try with res где надо
3)Вынес конкатенации где смог(в том месте не получается, не так рабоатет джоин стринги, как надо)
4)Убрал лишние аннотации

Separador commented on Home_12.2/src/main/java/DirectorRepositoryImpl.java:
```diff
@@ -76,32 +78,43 @@ public void delete(Director director) {
+            return listDirectors;
+        } else {
+            try {
+                throw new FileNotFoundException();
+            } catch (FileNotFoundException e) {
+                System.out.println("Empty list");
+                throw new RuntimeException(e);
```
Это какая-то странная штука, почему не бросить просто FileNotFoundException или сразу не бросить RuntimeException?

Separador commented on Home_13.4/src/test/java/RestAssuredTEst.java:
Зачем это тут?

Separador commented on Home_12.2/src/main/java/Launch.java:
Это кажется не нужно

Separador reviewed (changes requested):
1) Сейчас для каждого теста создается отдельное соединение к бд, что не очень хорошо. Лучше переиспользовать одно соединение
2) Удаление сохраненных данных в тестах лучше делать вне тестового метода
3) вызовы на подобии `movieRepository.get(1)` выглядят непонятно. Лучше вместо чисел завести константы с понятными названиями. А в тестах с режиссерами можно значение id брать из объекта для проверки
4) для читаемости, при получении значений из resultSet'а лучше пользоваться именами столбцов, а не номерами

methshest committed:
12.2
заделал по другому тесты, посмотри

Separador commented on Home_12.2/src/main/java/DirectorRepositoryImpl.java:
```diff
@@ -107,14 +111,14 @@ public List<Director> get(List<String> genres) {
-                System.out.println("Empty list");
-                throw new RuntimeException(e);
+                throw new RuntimeException();
+            } catch (RuntimeException e) {
```
Все равно не понимаю зачем тут catch

Separador commented on Home_12.2/src/main/java/MovieRepositoryImpl.java:
```diff
@@ -7,11 +7,7 @@ public class MovieRepositoryImpl implements MovieRepository {
-        } catch (SQLException e) {
-            throw new RuntimeException(e);
-        }
+        connection = DirectorRepositoryImpl.getConnection();
```
Переиспользование соединения таким образом неочень, но ладно
